Add a function to return the default ACL for a directory

aidanhs opened the pull request.

intgr commented:
Thanks, will try to take a look at this later today.

intgr commented on src/lib.rs:
```diff
@@ -337,6 +348,15 @@ impl PosixACL {
+    /// The caller *must not*:
+    /// - `acl_free` or otherwise invalidate the returned `acl_t`
+    /// - use this `acl_t` after the PosixACL has been dropped
+    pub fn as_raw(&self) -> acl_sys::acl_t {
```
It seems this method should be marked `unsafe`, what do you think?

Just wondering, what use case do you have for this? Maybe it makes sense to implement your needed feature directly in posix-acl, in addition to this.

aidanhs replied:
Moved discussion to other PR.

intgr commented on src/lib.rs:
```diff
@@ -188,14 +188,25 @@ impl PosixACL {
     }
 
-    /// Read a file's ACL and return as `PosixACL` object.
+    /// Read a path's ACL and return as `PosixACL` object.
```
:+1:

intgr commented on src/lib.rs:
```diff
@@ -188,14 +188,25 @@ impl PosixACL {
+        Self::read_acl_flags(path, acl_sys::ACL_TYPE_ACCESS)
+    }
+
+    /// Read a path's default ACL and return as `PosixACL` object.
```
Since this doesn't support files, how about "`Read a directory's default ACL ...`"

aidanhs replied:
Fixed

intgr commented:
Hopefully my CI configuration now works after #19. If you rebase/merge with latest master branch.

intgr commented:
Hi @aidanhs, are you still interested in finishing these? There's no rush, just wondering.

aidanhs commented:
I've split out the as_raw work to a separate PR and updated this one.

On feedback - I'm just really happy posix-acl exists so I don't have to create it myself! It hasn't got in my way at all, which I consider pretty important :smile:. The only thing that was a bit odd was the choice of SimpleError as an Error type - it's not an issue for my use-case, but others may prefer something more structured (`io::Error` could be an useful start, as POSIX ACL libraries look like they use the system errno rather than their own type).

intgr commented:
Merged, thanks!

> The only thing that was a bit odd was the choice of SimpleError as an Error type

That's a good point, this library started out as part of a CLI app [ego](https://github.com/intgr/ego) and it didn't occur to me to change error handling when I split it out.

I'll have to look into error handling patterns in Rust. `io::Error` does look like an option, indeed.

intgr commented:
I've opened issue #22 to track that progress.

intgr commented:
@aidanhs FYI I released version 0.4.0 with this feature and a bunch of other stuff.

aidanhs commented:
@intgr thanks!
